specs/merge/validator.md

```diff
@@ -60,9 +81,14 @@ All validator responsibilities remain unchanged other than those noted below. Na
 
 #### Constructing the `BeaconBlockBody`
 
-##### Execution Payload
+##### ExecutionPayload
 
-* Set `block.body.execution_payload = get_execution_payload(state, execution_engine, pow_chain)` where:
+To obtain an execution payload, the proposer of a block must take the following actions:
+
+1. Set `payload_id = prepare_execution_payload(state, pow_chain, fee_recipient, execution_engine)`, where:
+    * `state` is the state for which `is_proposer(state, validator_index)` returns `True`
```

Comment:
Any concern this `state` won't work well with `state.latest_execution_payload_header.block_hash`?

For post merge, normal payload, `state.latest_execution_payload_header.block_hash` changes on per block level. Where `is_proposer(state, validator_index)` can be known at the start of the epoch, and we also recommend calling `prepare_execution_payload` as soon as input parameters become known

Reply:
Right, but `is_proposer` call is bounded to the `state.slot` and in a normal case this state could become known only after the parent block is processed. It's not the state at the beginning of the epoch with artificially increased slot number. I will reformulate this statement to avoid any confusion.

Reply:
Updated the statement, please, let me know how does it read to you

Reply:
looks good to me